ports/slikenet/portfile.cmake

```
@@ -1,43 +1,54 @@
-    OUT_SOURCE_PATH SOURCE_PATH
-    REPO SLikeSoft/SLikeNet
-    REF cca394f05e9e9e3e315a85539e648f267d3f2fcc
-    SHA512 410954bda5a7be309eb71c3078f8ea67ff21aae2ce923f01db77b09265969f1350afb45b90194118bfad274f0a36a2d3bbc38d86a15507fdfc4bc8edc4a0204c
-    HEAD_REF master
-    PATCHES
-        fix-install.patch
-)
-#Uses an outdated OpenSSL version and is in an experimental namespace any way. As such we delete it here
-file(REMOVE_RECURSE "${SOURCE_PATH}/Source/src/crypto" "${SOURCE_PATH}/Source/include/slikenet/crypto")
-
-if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    set(SLIKENET_ENABLE_STATIC TRUE)
-    set(SLIKENET_ENABLE_DLL FALSE)
-else()
-    set(SLIKENET_ENABLE_STATIC FALSE)
-    set(SLIKENET_ENABLE_DLL TRUE)
-endif()
-
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
-    OPTIONS
-        -DSLIKENET_ENABLE_DLL=${SLIKENET_ENABLE_DLL}
-        -DSLIKENET_ENABLE_STATIC=${SLIKENET_ENABLE_STATIC}
-        -DSLIKENET_ENABLE_SAMPLES=FALSE
-)
-
-vcpkg_install_cmake()
-
-vcpkg_copy_pdbs()
-
-vcpkg_fixup_cmake_targets(CONFIG_PATH share/slikenet)
-
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include ${CURRENT_PACKAGES_DIR}/debug/share)
-
-configure_file("${CMAKE_CURRENT_LIST_DIR}/slikenet-config.cmake" "${CURRENT_PACKAGES_DIR}/share/slikenet/slikenet-config.cmake" COPYONLY)
-configure_file("${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" "${CURRENT_PACKAGES_DIR}/share/slikenet/vcpkg-cmake-wrapper.cmake" COPYONLY)
-file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
-file(INSTALL ${SOURCE_PATH}/license.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+vcpkg_fail_port_install(ON_TARGET "UWP")
+
+set(SLIKENET_PATCHES fix-install.patch)
+
+# We really don't want to apply this patch until we are not on MinGW:
+# - The patch disables the TwoWayAuthentication plugin which has problems on MinGW
+#   Not a part of the library core, so we just disable it on MinGW
+# - The patch uses third-party implementations of inet_pton/inet_ntop
+#   License also have been patched to contain third-party copyrights
+if(VCPKG_TARGET_IS_MINGW)
+    list(APPEND SLIKENET_PATCHES fix-mingw-build.patch)
+endif()
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO SLikeSoft/SLikeNet
+    REF 3b3632a4cd4fbe3d5f19d7a8b1e1e69c663e8cee
+    SHA512 62fa0e00d1763faf0226d8fb94eb3472adef993ee72c081d25089f780c2df0a4db5b3ce3a91d3d1714c019d5c4dfc6a87024430574cefca1f5e56c238a0d3ec8
+    HEAD_REF master
+    PATCHES ${SLIKENET_PATCHES}
+)
+
+# Uses an outdated OpenSSL version and is in an experimental namespace any way. As such we delete it here
+file(REMOVE_RECURSE "${SOURCE_PATH}/Source/src/crypto" "${SOURCE_PATH}/Source/include/slikenet/crypto")
+
+if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    set(SLIKENET_ENABLE_STATIC TRUE)
+    set(SLIKENET_ENABLE_DLL FALSE)
+else()
+    set(SLIKENET_ENABLE_STATIC FALSE)
+    set(SLIKENET_ENABLE_DLL TRUE)
+endif()

```

```suggestion
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" SLIKENET_ENABLE_STATIC)
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" SLIKENET_ENABLE_DLL)
```
Please see the reference here https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries